### ISSUE TITLE
FR: Allow submitting content from ... menu on posts screen

### ISSUE THREAD

**Issue by scstraus** [aeharding/voyager#730]
Hi there. Let me begin by saying I love Voyager and am very appreciative that I have such a nice UI for Lemmy on my mobile. I have just one request for how it handles submitting posts. 

Actually for a long time I thought it didn't have the feature to submit posts at all because I couldn't find it. It wasn't until I read in the release notes that it had this feature that I had to go digging around everywhere to find it. It took me a couple weeks of keeping my eyes peeled to find it in the community page.

I gave doing it this way a chance, but compared to the way some of the other apps do it, I find it rather hard to use. The issue for me is that it the community search is rather hard to use. I will compare the process on your app to the process on liftoff! and hopefully you will see why I like their submission process better.

Process on Voyager:

1) Press the Search button
2) Select Community search
3) Enter search term like "videos"
4) Click the sort button
5) Sort by "top"
6) Sort by "all time"
7) Click the community name
8) Click the ... menu
9) Click Submit Post
10) Fill out post details
11) Click Post
12) Click the sort button on the posts screen
13) Choose "hot" again

Process on Liftoff!:
1) Press + button on home (equivilent to communities) screen
2) Fill out post details (community search is here and sorts by top of all time making it easy to find the right community)
3) Click Post

Hopefully that expresses the differences in the experiences.

**Comment by aeharding** [comment]
Hi there! Yeah, there is definitely more friction here but it's implemented the way Apollo used to have it implemented.

The plus button is not straightforward to implement because it would conflict with the customizable hide posts button.

So there's more research needed on this.

**Comment by slrgt** [comment]
Reposting the mockup from my post here.

![image](https://github.com/aeharding/voyager/assets/114287507/028605c8-2a1d-4454-a305-9cd6d4a0d6e3)

Voyager can mix the setting snd profile page so the middle button on the navbar is a + button. It is much more important for people to be able to post new content to Lemmy than view their old posts. Voyager shouldn't limit itself to how Apollo did things since there's many things that Apollo still needed improvements in and this is a big one.

Thanks for the amazing work on Voyager!

**Comment by scstraus** [comment]
Fully agree, it's long past the time where this feature should be implemented. I keep a second app, Arctic, only to submit content since it's so broken on Voyager. Is there any updates here?

**Comment by scstraus** [comment]
> Hi there! Yeah, there is definitely more friction here but it's implemented the way Apollo used to have it implemented.
> 
> The plus button is not straightforward to implement because it would conflict with the customizable hide posts button.
> 
> So there's more research needed on this.

FWIW, I don't need a plus button. Arctic hides this in the ... menu on the main page, and that's perfectly fine with me.

**Comment by aeharding** [comment]
I'm a bit confused by your steps on voyager. Here's a comparison:

Process on Voyager:

    Search for a community (quick switcher)
    Press "..."
    Press "Create Post"
    Fill out post details
    Click Post
    

Process on Arctic:

    Press "..."
    Press "Create Post"
    Search for a community
    Fill out post details
    Click Post

It's the same number of steps, the only difference is when in the process you search for the community.

**Comment by scstraus** [comment]
Not at all. The process on Arctic matches exactly the process on Liftoff! except for the fact that you have to go into a submenu to get to it, so technically one more click. 

All of these steps exist in the Voyager submission that do not exist in the Arctic submission:
Process on Voyager:

    Press the Search button
    Select Community search
    Enter search term like "videos"
    Click the sort button
    Sort by "top"
    Sort by "all time"
    Click the community name
    Click the sort button on the posts screen
    Choose "hot" again

But maybe the issue is better reframed as that the community search is broken for posting. When I do it with arctic, filling out the community is a 1 second operation because the community search brings up the largest communities first, with Voyager (at least the last time I checked), the sorting made it very time consuming to find the correct community, and I have to change the sorting to "top" to find anything, then I have to change the search back to "hot"to actually view posts in a useful way. So fixing that would get you part of the way there, but it's still far more convenient to have this on the front page rather than having to find a community and go into it's page to get there, because then you have to navigate back out when you are done (also an extra step I forgot to include in my description, so you can add that one).

**Comment by aeharding** [comment]
Hey, I am still trying to understand what is going on because I don't fully understand why you're going through so many steps to submit a post. I don't understand this:

```
Press the Search button
Select Community search
Enter search term like "videos"
Click the sort button
Sort by "top"
Sort by "all time"
Click the community name
Click the sort button on the posts screen
Choose "hot" again
```

Why not just:

```
Tap quick switcher from all/home/local
Type term
Press enter
```

> the sorting made it very time consuming to find the correct community, and I have to change the sorting to "top" to find anything

OK, that seems to be the root of your issue. However the quick switcher sorts by TopAll (after your subscriptions).
